File: identity-service/src/main/java/com/example/identity_service/dto/response/OrderResponse.java
package com.example.identity_service.dto.response;


import com.example.identity_service.entity.Customer;
import com.example.identity_service.entity.Employee;
import com.example.identity_service.entity.OrderDetail;
import com.example.identity_service.enums.OrderStatus;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class OrderResponse {
    Integer orderId;

    Integer customerId;

    LocalDateTime orderDate;

    BigDecimal totalAmount;

    OrderStatus status;

    List<OrderDetail> orderDetails;
}

File: computer-selling-v1/src/app/model/response/order.response.ts
import { OrderDetail } from "./orderDetail.response"
import { Payment } from "./payment.response"
import { Shipment } from "./shipment.response"

export interface Order {
    orderId: number
    orderDate: string
    totalAmount: number
    status: string
    orderDetails: OrderDetail[]
    customerId: number
    payment: Payment
    shipmentAddress: Shipment
}